### PR TITLE
fix(cli): fixed weapp component file watch

### LINKED ISSUE #2380: Cannot read property 'file' of undefined
引入一个原生的组件，修改其中的js文件，就会报错
```
UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'file' of undefined
    at Compile.<anonymous> (/small2/node_modules/@wepy/cli/core/plugins/parser/file.js:17:27)
    at Compile.hookUnique (/small2/node_modules/@wepy/cli/core/hook.js:54:21)
```

### PR DESCRIPTION
**Pull request by dlhandsome**
##### Checklist

- [ ] `npm run test` passes
- [ ] tests and/or benchmarks are included
- [ ] cases or donate is changed or added
- [ ] documentation is changed or added

Fixed #2380

### CODE REVIEW

File: packages/cli/core/plugins/parser/component.js [hunk 1]
```diff
@@ -37,12 +37,19 @@ exports = module.exports = function () {
...
 
-    if (!this.compiled[file]) {
-      this.compiled[file] = {};
-    }
     let wpyTask = [];
 
-    ['.js', '.wxml', 'wxss', '.json'].forEach(v => this.fileDep.addDeps(file + v));
+    ['.js', '.wxml', 'wxss', '.json'].forEach(v => {
+      const weappFile = file + v;
+      this.fileDep.addDeps(weappFile)
```
Comment (dlhandsome) [inline packages/cli/core/plugins/parser/component.js]: missing semicolon

File: packages/cli/core/plugins/parser/component.js [hunk 2]
```diff
@@ -37,12 +37,19 @@ exports = module.exports = function () {
...
+
+      if (!this.compiled[weappFile]) {
+        this.compiled[weappFile] = {
```
Comment (Gcaufy) [inline packages/cli/core/plugins/parser/component.js]: 意思是 weapp component 建了三份 context。而且内容都是一样的？除了 file ?

Comment (Gcaufy) [reply packages/cli/core/plugins/parser/component.js]: ### 四份

Comment (dlhandsome) [reply packages/cli/core/plugins/parser/component.js]: 是的

**Pull request closed without merging**